exporters/metrics/src/main/java/com.google.cloud.opentelemetry.metric/MetricExporter.java

```diff
@@ -0,0 +1,201 @@
+      MetricServiceClient metricServiceClient) {
+    return new MetricExporter(projectId, metricServiceClient);
+  }
+
+  private static MetricExporter createWithCredentials(
+      String projectId,
+      Credentials credentials,
+      Duration deadline) throws IOException {
+    MetricServiceSettings.Builder builder =
+        MetricServiceSettings.newBuilder()
+            .setCredentialsProvider(
+                FixedCredentialsProvider.create(checkNotNull(credentials, "Credentials not provided.")));
+    builder.createMetricDescriptorSettings()
+        .setSimpleTimeoutNoRetries(org.threeten.bp.Duration.ofMillis(deadline.toMillis()));
+    return new MetricExporter(projectId, MetricServiceClient.create(builder.build()));
+  }
+
+  @Override
+  public ResultCode export(Collection<MetricData> metrics) {
+    List<TimeSeries> allTimesSeries = new ArrayList<>();
+
+    for (MetricData metricData : metrics) {
+      MetricDescriptor descriptor = mapMetricDescriptor(metricData);
+      if (descriptor == null) {
+        continue;
+      }
+      metricServiceClient.createMetricDescriptor(
+          CreateMetricDescriptorRequest.newBuilder().setName(PROJECT_NAME_PREFIX + projectId)
+              .setMetricDescriptor(descriptor)
+              .build());
+
+      MetricWithLabels updateKey = new MetricWithLabels(descriptor.getType(),
+          metricData.getDescriptor().getConstantLabels());
+
+      // We are expecting one point per MetricData
+      if (metricData.getPoints().size() != 1) {
+        logger.error("There should be exactly one point in each metricData, found {}",
+            metricData.getPoints().size());
+        continue;
+      }
+      MetricData.Point metricPoint = metricData.getPoints().iterator().next();
+
+      // Cloud Monitoring API allows, for any combination of labels and
+      // metric name, one update per WRITE_INTERVAL seconds
+      long pointCollectionTime = metricPoint.getEpochNanos();
+      if (lastUpdatedTime.containsKey(updateKey)
+          && pointCollectionTime <= lastUpdatedTime.get(updateKey) / NANO_PER_SECOND + WRITE_INTERVAL_SECOND) {
+        continue;
+      }
+
+      Metric metric = mapMetric(metricData, descriptor.getType());
+      Point point = mapPoint(metricData, metricPoint, updateKey, lastUpdatedTime);
+      if (point == null) {
+        continue;
+      }
+
+      allTimesSeries.add(TimeSeries.newBuilder()
+          .setMetric(metric)
+          .addPoints(point)
+          .setResource(MonitoredResource.newBuilder().build())
+          .setMetricKind(descriptor.getMetricKind())
+          .build());
+    }
+    createTimeSeriesBatch(metricServiceClient, ProjectName.of(projectId), allTimesSeries);
+    return ResultCode.SUCCESS;
+  }
+
+  private static void createTimeSeriesBatch(MetricServiceClient metricServiceClient, ProjectName projectName,
+      List<TimeSeries> allTimesSeries) {
+    List<List<TimeSeries>> batches = Lists.partition(allTimesSeries, MAX_BATCH_SIZE);
```

Comment:
It's possible that `allTimesSeries.isEmpty()` with the logic above. Will that be handled correctly here?
May just add a check and return.